pkg/ciliumenvoyconfig/reconciler.go

```diff
@@ -90,6 +170,23 @@ func (ops *envoyOps) Update(ctx context.Context, txn statedb.ReadTxn, _ statedb.
 	}
 
 	err := ops.xds.UpdateEnvoyResources(ctx, prevResources, resources)
+
+	if err != nil && isPortBindingError(err) {
+		ops.log.Warn("Port binding failed, attempting to reallocate ports and retry",
+			logfields.Error, err)
+
+		retryResources := resources
+		retryErr := ops.retryWithNewPorts(ctx, prevResources, retryResources)
+		if retryErr == nil {
+			resources = retryResources
```

Comment:
I'm not sure if update of `resources` is allowed here, @joamaki could you chime in here? Is `res.Resources` here considered read-only, i.e., is it stored in statedb? If yes, what would be the right way to retry with a new proxy port number? If not, then the shallow-copies here and cloning of the listener in `retryWithNewPorts` would not be necessary.